rfc/rfc-0006-child-resource-names.md

```diff
@@ -0,0 +1,43 @@
+# RFC-0006: Child Resource Names
+
+**Authors:** Scott Andrews
+
+**Status:**
+
+**Pull Request URL:** [#1369](https://github.com/projectriff/riff/pull/1369)
+
+**Superseded by:** N/A
+
+**Supersedes:** N/A
+
+**Related:** N/A
+
+
+## Problem
+When a controller creates a new resource, it need to either pick a static name, or let kubernetes generate a name. Static names are predictable, but may already be in use. Generated names cannot be predicated, but are guaranteed to be available.
+
+### Anti-Goals
+This RFC does not prescribe which naming scheme to use for any specific resource.
+
+## Solution
+While there will not be a cut and dry answer for each situation, there are heuristics to guide the decision:
+- if a resource is not consumed directly by a non-controlled resource by name, generate the name (e.g. Pods in a ReplicaSet)
+- if the resource can be queried via a label selector instead of by name, generate the name (e.g. Pods fronted by a Service)
+- if a resource's name surfaces in outward facing way that is part of the public API, consider using a static name (e.g. Services in DNS)
+- if the name needs to be stable between different instances of the same resource, use a static name (e.g. immutable resources that cannot be updated)
+- generate the name
```

Comment:
```suggestion
- if none of the above criteria apply, generate the name
```

Reply:
If the catch-all is to generate the name, then each of the other criteria should be about when to use a static name.

Reply:
@glyn the general idea is that we should favor using generated names. Even if the item that leads to choosing a static name is true, there may be a superseding condition that says the resource should still use a generated name. For example, not all Services should us static names, even thought the name of a service is used to expose a DNS record that can be consumed by other resources.